src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -217,4 +258,135 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String queryId() {
         return NAME;
     }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String field;
+        FieldMapper mapper = parseContext.fieldMapper(fieldName);
+        if (mapper != null) {
+            field = mapper.names().indexName();
+        } else {
+            field = fieldName;
+        }
+
+        Analyzer analyzerObj;
+        if (analyzer == null) {
+            if (mapper != null) {
+                analyzerObj = parseContext.getSearchAnalyzer(mapper);
+            } else {
+                analyzerObj = parseContext.mapperService().searchAnalyzer();
+            }
+        } else {
+            analyzerObj = parseContext.mapperService().analysisService().analyzer(analyzer);
+            if (analyzerObj == null) {
+                throw new IllegalArgumentException("no analyzer found for [" + analyzer + "]");
+            }
+        }
+        
+        Occur highFreqOccur = highFreqOperator.toMustOrShouldClause();
+        Occur lowFreqOccur = lowFreqOperator.toMustOrShouldClause();
+
+        ExtendedCommonTermsQuery query = new ExtendedCommonTermsQuery(highFreqOccur, lowFreqOccur, cutoffFrequency, disableCoord, mapper);
+        parseQueryString(query, text, field, analyzerObj, lowFreqMinimumShouldMatch, highFreqMinimumShouldMatch);
+
+        query.setBoost(boost);
```

Comment:
just to be sure, I woud move this set two lines up, between the query assignment and the parseQueryString call, there query will never be null ( I know that with your change it's not null anymore, but we have to look at what the method does to be sure).